egui_extras/src/table.rs

```diff
@@ -0,0 +1,280 @@
+                    clicked: false,
+                };
+                header(row);
+            }
+        }
+
+        Table {
+            ui,
+            padding: self.padding,
+            widths,
+            scroll: self.scroll,
+            striped: self.striped,
+        }
+    }
+
+    /// Create table body without a header row
+    pub fn body<F>(self, body: F)
+    where
+        F: for<'b> FnOnce(TableBody<'b>),
+    {
+        let widths = self.sizing.into_lengths(
+            self.ui.available_rect_before_wrap().width() - 2.0 * self.padding.outer,
+            self.padding.inner,
+        );
+
+        Table {
+            ui: self.ui,
+            padding: self.padding,
+            widths,
+            scroll: self.scroll,
+            striped: self.striped,
+        }
+        .body(body)
+    }
+}
+
+pub struct Table<'a> {
+    ui: &'a mut Ui,
+    padding: Padding,
+    widths: Vec<f32>,
+    scroll: bool,
+    striped: bool,
+}
+
+impl<'a> Table<'a> {
+    /// Create table body after adding a header row
+    pub fn body<F>(self, body: F)
+    where
+        F: for<'b> FnOnce(TableBody<'b>),
+    {
+        let padding = self.padding;
+        let ui = self.ui;
+        let widths = self.widths;
+        let striped = self.striped;
+        let start_y = ui.available_rect_before_wrap().top();
+        let end_y = ui.available_rect_before_wrap().bottom();
+
+        egui::ScrollArea::new([false, self.scroll]).show(ui, move |ui| {
+            let layout = Layout::new(ui, padding, LineDirection::TopToBottom);
+
+            body(TableBody {
+                layout,
+                widths,
+                striped,
+                odd: true,
+                start_y,
+                end_y,
+            });
+        });
+    }
+}
+
+pub struct TableBody<'a> {
+    layout: Layout<'a>,
+    widths: Vec<f32>,
+    striped: bool,
+    odd: bool,
+    start_y: f32,
+    end_y: f32,
+}
+
+impl<'a> TableBody<'a> {
+    /// Add rows with same height
+    /// Is a lot more performant than adding each individual row as non visible rows must not be rendered
+    pub fn rows(mut self, height: f32, rows: usize, mut row: impl FnMut(usize, TableRow)) {
+        let delta = self.layout.current_y() - self.start_y;
+        let mut start = 0;
+
+        if delta < 0.0 {
+            start = (-delta / height).floor() as usize;
+
+            let skip_height = start as f32 * height;
+            TableRow {
+                layout: &mut self.layout,
+                widths: self.widths.clone(),
+                striped: self.striped && self.odd,
+                height: skip_height,
+                clicked: false,
+            }
+            .col(|_| ()); // advances the cursor
+        }
+
+        let max_height = self.end_y - self.start_y;
+        let count = (max_height / height).ceil() as usize;
+        let end = cmp::min(start + count, rows);
+
+        if start % 2 != 0 {
+            self.odd = false;
+        }
```

Comment:
surely we should always assign to `self.odd` ?

```suggestion
        self.odd = start % 2 == 1;
```